wlauto/workloads/googlephotos/__init__.py

```diff
@@ -0,0 +1,107 @@
+#    Copyright 2014-2016 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import wlauto.common.resources
+
+from wlauto import AndroidUiAutoBenchmark, Parameter
+from wlauto.exceptions import ValidationError
+from wlauto.utils.types import list_or_string
+
+__version__ = '0.1.1'
+
+
+class Googlephotos(AndroidUiAutoBenchmark):
+
+    name = 'googlephotos'
+    package = 'com.google.android.apps.photos'
+    activity = 'com.google.android.apps.photos.home.HomeActivity'
+    view = [package + '/com.google.android.apps.consumerphotoeditor.fragments.ConsumerPhotoEditorActivity',
+            package + '/com.google.android.apps.photos.home.HomeActivity',
+            package + '/com.google.android.apps.photos.localmedia.ui.LocalPhotosActivity',
+            package + '/com.google.android.apps.photos.onboarding.AccountPickerActivity',
+            package + '/com.google.android.apps.photos.onboarding.IntroActivity']
+    description = """
+    A workload to perform standard productivity tasks with Google Photos.  The workload carries out
+    various tasks, such as browsing images, performing zooms, post-processing and saving a selected
+    image to file.
+
+    Test description:
+    1. Four images are copied to the device
+    2. The application is started in offline access mode
+    3. Gestures are performed to swipe between images and pinch zoom in and out of the selected
+       image
+    4. The Colour of a selected image is edited by selecting the colour menu, incrementing the
+       colour, resetting the colour and decrementing the colour using the seek bar.
+    5. A Crop test is performed on a selected image.  UiAutomator does not allow the selection of
+       the crop markers so the image is tilted positively, reset and then negatively to get a
+       similar cropping effect.
+    6. A Rotate test is performed on a selected image, rotating anticlockwise 90 degrees, 180
+       degrees and 270 degrees.
+    """
+
+    parameters = [
+        Parameter('dumpsys_enabled', kind=bool, default=True,
+                  description="""
+                  If ``True`` turns on the action logger which outputs
+                  timestamps to logcat for actions recorded in the workload.
+                  """),
+        Parameter('test_images', kind=list_or_string, mandatory=True,
```

Comment:
Why is this a mandatory parameter?

Presumably, images used will have a significant impact on the measurements collected; results collected using two different image sets would not be comparable. Therefore the images are an intrinsic part of the test, rather than an input into it. 

While I can see someone wanted to use this to examine issues with specific images, so it is useful to have the *option* to override the default set your own, the default should be provided. Moreover, any results collected using something other than the images that are part of the tests, need to be identified as such in the output.